packages/tx/test/eip7702.spec.ts

```diff
@@ -0,0 +1,201 @@
+              yParity: '0x1',
+              r: ones32,
+              s: ones32,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 'nonce list should consist of at most 1 item')
+
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
+              address: `0x${'20'.repeat(20)}`,
+              nonce: ['0x1'],
+              yParity: '0x1',
+              r: ones32,
+              s: undefined as never,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 's is not defined')
+
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
+              address: `0x${'20'.repeat(20)}`,
+              nonce: ['0x1'],
+              yParity: '0x1',
+              r: undefined as never,
+              s: ones32,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 'r is not defined')
+
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
+              address: `0x${'20'.repeat(20)}`,
+              nonce: ['0x1'],
+              yParity: undefined as never,
+              r: ones32,
+              s: ones32,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 'yParity is not defined')
+
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
+              address: `0x${'20'.repeat(20)}`,
+              nonce: undefined as never,
+              yParity: '0x1',
+              r: ones32,
+              s: ones32,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 'nonce is not defined')
+
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
+              address: undefined as never,
+              nonce: ['0x1'],
+              yParity: '0x1',
+              r: ones32,
+              s: ones32,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 'address is not defined')
+
+    assert.throws(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: undefined as never,
+              address: `0x${'20'.repeat(20)}`,
+              nonce: ['0x1'],
+              yParity: '0x1',
+              r: ones32,
+              s: ones32,
+            },
+          ],
+        },
+        { common }
+      )
+    }, 'chainId is not defined')
+
+    assert.doesNotThrow(() => {
+      EOACodeEIP7702Transaction.fromTxData(
+        {
+          authorizationList: [
+            {
+              chainId: '0x',
```

Comment:
This is valid? To have the `chainId` set to `0x`??

Reply:
Respectively going further (without having read the EIP): shouldn't there be a check on our side if the chainId here matches the one from Common and otherwise (we) throw?

Reply:
Yes, 0x means it is valid on all chains. From the EIP:

`2. Verify the chain id is either 0 or the chain’s current ID.`

Reply:
Ah, interesting.

Reply:
Ok, but then I guess at least Common (and therefore signed tx through chain ID in signature) and chain ID here differing should throw on the tx side?

Because atm it does not, tested with mainnet Common and chain ID 2.

Reply:
Had another look into the EIP, this (chain ID check) is only mentioned in execution, but would this make sense to allow such a tx in the network (so: in a block?).

If you haven't heard anything specific about that we should likely ask for confirmation/explanation/re-evaluation.

Reply:
Yes, it is thus a valid tx and it is thus allowed in the block, we should not throw. If the chain id does not match, we thus "skip" the tuple and do not invalidate the tx.

Reply:
What was the reasoning to allow this during the EIP discussion/specification?

Reply:
(or: *was* this a decision?)

Reply:
I don't know the exact reason, but I think it is among the lines just like with access lists, you are allowed to put the same address twice. This does not invalidate the tx, you just have to pay more gas (so no one wil do it). The same reasoning works here, you can input any authority tuple, but if it does not match the chain id it will do absolutely nothing, but you have to pay gas.